### ISSUE TITLE
[HOLD for payment 2023-08-17] [$1000] Update Task title/description/assignee when offline is not grey out

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#23200]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to any Task report
2. Turn off network
3. Update task title/description/assignee
## Expected Result: 
Task title/description/assignee should be grey out when update in offline mode
## Actual Result: 
Task title/description/assignee is not grey out

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.43-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/25c616e9-afef-47d9-a889-8a6a7b62e175

![Snip - (4) New Expensify - Google Chrome](https://github.com/Expensify/App/assets/43996225/503a1dbe-6865-45c7-8327-576cee9b5d35)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @hoangzinh
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689611035906189

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0138286787014f7dd4</li>
        <li>Upwork Job ID: 1682219760636948480</li>
        2023-07-21<li>Automatic offers: </li>
<ul>
<li>mollfpr | Reviewer | 25750552</li>
<li> |  | 0</li>
<li>hoangzinh | Reporter | 25750553</li>
</ul></li>
    </ul>
</details>

**Comment by tjferriss** [comment]
It actually looks like it's intended that we do not grey this out per https://github.com/Expensify/App/issues/22907

**Comment by tjferriss** [comment]
Got it. Thanks for the clarification. I'm able to reproduce the bug.

**Comment by mollfpr** [comment]
I agree with @allroundexperts RCA and solution. We didn't use the `OfflineWithFeedback` on the desired fields.

Let's go with go @allroundexperts [proposal](https://github.com/Expensify/App/issues/23200#issuecomment-1642782587) 🚀

🎀 👀 🎀 C+ reviewed!

**Comment by neil-marcellini** [comment]
> Let's go with go @allroundexperts [proposal](https://github.com/Expensify/App/issues/23200#issuecomment-1642782587) 🚀

Yeah this looks good. A couple notes so far:
- We don't need to pass optional props unless we're going to use them
- There might be errors from the backend under `report.errorFields.editTask`

cc @thienlnam in case you want to also review the PR.

**Comment by thienlnam** [comment]
Thanks for the ping - agree with the solution

**Comment by tjferriss** [comment]
Some discussion on the error placement in the PR. Otherwise, it looks like we're on track to get this merged.

**Comment by Pujan92** [comment]
@allroundexperts @tjferriss [This PR](https://github.com/Expensify/App/pull/23705) seems to causing a regression 
https://expensify.slack.com/archives/C049HHMV9SM/p1691084126708399

**Comment by tjferriss** [comment]
@hoangzinh and @mollfpr are both hired in Upworks: https://www.upwork.com/ab/applicants/1682219760636948480/hired. We're waiting on the regression period for payment. 

@mollfpr @allroundexperts can you get started on the checklist?

**Comment by mollfpr** [comment]
> [@mollfpr / @allroundexperts] The PR that introduced the bug has been identified. Link to the PR:
 [@mollfpr / @allroundexperts] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

I don't think there's an offending PR. We haven't implemented the offline feedback in the first place.

> [@mollfpr / @allroundexperts] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

The regression step should be enough.

>[@mollfpr / @allroundexperts] Determine if we should create a regression test for this bug.
 [@mollfpr / @allroundexperts] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

1. Go to any Task report
2. Turn off the network
3. Update task title/description/assignee
4. Verify that the updated fields are greyed out
5. Turn on the network
6. **Verify** that fields are not greyed out
7. Turn off the network again
8. **Verify** that fields are not greyed out

**Comment by michaelhaxhiu** [comment]
Should this PR have addressed the problem cited in https://github.com/Expensify/App/issues/25956#issuecomment-1699487935? They feel like super similar / duplicate style bugs, so I'm curious if we can update the PR (or draft a new PR) to accommodate this too?

**Comment by mollfpr** [comment]
@michaelhaxhiu My test on the PR shows the assignee field grey out but not in the current `main`.

I think https://github.com/Expensify/App/issues/25956 is the legit report. The issue is a regression from https://github.com/Expensify/App/pull/22778. In this issue PR, we are still using `editTaskAndNavigate` and have the RBR action set up there. In #22778, we changed that to `editTaskAssigneeAndNavigate`, but it hasn't got the RBR action.

**Comment by michaelhaxhiu** [comment]
Thanks luthfi @mollfpr i appreciate your input as always 🙏

**Comment by tjferriss** [comment]
Regression test submitted to Applause.
